### DISCUSSION TITLE
LDAP_SERVERS syntax

### DISCUSSION THREAD

**Discussion by lieutenant-nemo** [usnistgov/NEMO#241] [Q&A] [answered]
Hey there,
first of all: great work!
I´m quite confused about the right syntax for LDAP_SERVERS in settings.py. If I use "url", "port", "use_ssl", "bind_as_authentication" and "base_dn" (as it is in the original setting.py but with other values), I got:
"The user was denied access: LDAPInvalidDNSyntaxResult - 34 - invalidDNSyntax - None - invalid DN - bindResponse - None"
I have tried other things also, like "bind_dn" and the like, but without success.
Any advice? What is the right syntax?
I would appreciate any help or comment.
Thanx!

**Comment by rptmat57** [comment]
make sure there are no spaces in the `base_dn` attribute: `"dc=example,dc=com"`?

**Reply by lieutenant-nemo** [reply]
Hi,
thank you for your answer. There are no spaces.
The question is: which syntax should I use if I need to specify a bind dn or a bind password? 
Which are allowed?

**Comment by rptmat57** [comment, marked as answer]
you would set `bind_username` and `bind_password`. there is no bind_dn option

**Reply by lieutenant-nemo** [reply]
Hi again,
it worked!
I tought `bind_username` was just the username but you have to write the whole chain, in our case "uid=,cn=,dc=,dc=".
Thak you very much!
